### DISCUSSION TITLE
[Suggestion] Add attribute to solve ambiguous things btw languages

### DISCUSSION THREAD

**Discussion by dayo05** [dotnet/csharplang#6410] [Language Ideas] [answered]
Dotnet can execute programs which compiled from various kinds of languages. But some case it cause ambiguous.
For example, In C#
```cs
public static void Func(int[] a) {}
public static void Func(long[] a) {}
```
We can call this method in C# but in case of F#, int[] and long[] can implicit converted so it cause ambiguous. 
So I suggests creation of attribute like `[OnlyIn("csharp")]` or `[IgnoreOn("fsharp")]` to tell compiler to ignore some method.

Thanks for reading :D

**Comment by svick** [comment, marked as answer]
> We can call this method in C# but in case of F#, int[] and long[] can implicit converted so it cause ambiguous.

Can you explain that in more detail? [From a quick F#-only test, it seems to me there is no ambiguity.](https://sharplab.io/#v2:DYLgZgzgNALiBuBTATgSzATygExAagB8YMAHRAAgGFyBeAWAChznyIYBDGVAY3IFtEfAEYpyAWQAU7EOVQA7GAGYATAG0AugEpa5CZsYtWHLrwHDRk6bIUA2ACwbtNXfqYtGlAHSTVBcneVyAi0Pbwlff2VgIJCGLx8/AIAZGM0gA===)

> So I suggests creation of attribute like `[OnlyIn("csharp")]` or `[IgnoreOn("fsharp")]` to tell compiler to ignore some method.

I don't think this is sustainable. What you're describing is a regular .Net API. If F# has issues with it, then that should be fixed in F#.

**Comment by ufcpp** [comment]
Similar to https://github.com/dotnet/csharplang/discussions/964

**Comment by dayo05** [comment]
Ah, I found that this problem has not caused by difference by language!  Thanks for your help :D
